cpp/src/Ice/ConnectionI.cpp

```diff
@@ -1917,7 +1917,7 @@ Ice::ConnectionI::ConnectionI(
       _asyncRequestsHint(_asyncRequests.end()),
       _messageSizeMax(connector ? _instance->messageSizeMax() : adapter->messageSizeMax()),
       _batchRequestQueue(new BatchRequestQueue(instance, endpoint->datagram())),
-      _readStream(_instance.get(), Ice::currentProtocolEncoding),
+      _readStream(instance.get()),
```

Comment:
We don't have to pass `currentProtocolEncoding` because this is the default, right?

Reply:
Yes, but most importantly, the encoding of an InputStream over an empty does not / should not matter. We are obviously not decoding from these streams. They are constructed in a few places and as far I can tell, it's just to swap them with other real InputStreams.

That's unlike the encoding of an OutputStream, which actually matters. An OutputStream has an expandable buffer.